storagemarket/deal_handler.go

```diff
@@ -37,3 +36,27 @@ func (d *dealHandler) subscribeUpdates() (event.Subscription, error) {
 	}
 	return sub, nil
 }
+
+func (dh *dealHandler) cancel() error {
+	dh.transferMu.Lock()
+	defer dh.transferMu.Unlock()
+
+	if dh.transferFinished {
+		return dh.transferErr
+	}
+
+	dh.transferCancel()
+
+	select {
+	case err := <-dh.transferDone:
+		dh.transferFinished = true
+		dh.transferErr = err
+		return err
+	case <-dh.providerCtx.Done():
+		return dh.providerCtx.Err()
```

Comment:
If the provider context has been shutdown, the deal will get cancelled anyway, effectively cancelling the transfer. So in this case I think we should return nil, as the cancel action has succeeded.

Reply:
done.